packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -763,18 +763,136 @@ String _debugCompareFloats(String labelA, double valueA, String labelB, double v
          'apply the min() or max() functions, or the clamp() method, to the $labelB? ';
 }
 
-// ///
-// /// ## Writing a RenderSliver subclass
-// ///
-// /// ### Painting
-// ///
-// /// The [paint] method is called with an [Offset] to the top-left corner of the
-// /// sliver, _regardless of the axis direction_.
-// ///
-// /// ### childScrollOffset
-// ///
-// /// If the subclass positions children anywhere other than at scroll offset
-// /// 0.0, you need to override [childScrollOffset]...
+/// Base class for the render objects that implement scroll effects in viewports.
+///
+/// A [RenderViewport] has a list of child slivers. Each sliver — literally a
+/// thin piece of the viewport's contents — is laid out in turn, so as to cover
+/// the viewport.
+///
+/// Slivers participate in the _sliver protocol_, which is essentially a 15+
+/// dimension coordinate system wherein during [layout] each sliver receives a
+/// [SliverConstraints] object and computes a corresponding [SliverGeometry]
+/// that describes where it fits in the viewport. (The "15+ dimensions" refers
+/// to the number of properties found on [SliverConstraints] and
+/// [SliverGeometry]. The number is not precise because some of the properties
+/// are interdependent.)
```

Comment:
I don't think the +15 language is that helpful.  It makes me think this is too complicated to understand.  I'd just say the more concrete thing that they get a SliverConstraints and produce a SliverGeometry, perhaps giving the analogy to BoxConstraints and Size.